rmw_zenoh_cpp/src/detail/liveliness_utils.cpp

```diff
@@ -679,7 +678,18 @@ std::string demangle_name(const std::string & input)
 }  // namespace liveliness
 
 ///=============================================================================
-size_t hash_gid(const uint8_t gid[RMW_GID_STORAGE_SIZE])
+size_t hash_gid(const std::vector<uint8_t> gid)
```

Comment:
I'm not a big fan of having both `hash_gid` and `hash_gid_p`.  Is there any way we can just do one of these?  Barring that, can we make one call the other?